WooCommerce/Classes/ViewRelated/Products/Media/ProductImagesSaver.swift

```diff
@@ -75,13 +75,31 @@ private extension ProductImagesSaver {
                 self.uploadStatusesSubscription = nil
             }
             stores.dispatch(action)
-        case .variation(_, _):
-            // TODO: 7021 - update variation images action with a different endpoint
-            return
+        case .variation(let productID, let variationID):
+            // Product variation only has up to one image.
+            guard let image = images.first else {
+                return
+            }
+            let action = ProductVariationAction.updateProductVariationImage(siteID: siteID,
+                                                                            productID: productID,
+                                                                            variationID: variationID,
+                                                                            image: image) { [weak self] result in
+                guard let self = self else { return }
+                switch result {
+                case .success(let productVariation):
+                    onProductSave(.success(productVariation.image.map { [$0] } ?? []))
+                case .failure(let error):
+                    onProductSave(.failure(error))
+                }
+                self.imageStatusesToSave = []
+                self.assetUploadSubscription = nil
+                self.uploadStatusesSubscription = nil
```

Comment:
Could we create a helper method within the following function
```swift
func saveProductImages(_ images: [ProductImage], onProductSave: @escaping (Result<[ProductImage], Error>) -> Void)
```
for resetting values and reuse the new helper function for both `.product` and `.variation` cases?

Reply:
yea good idea! updated in https://github.com/woocommerce/woocommerce-ios/pull/7195/commits/981ae81ad0b4ab079f5920aa8506fc680e873c58